### PR TITLE
Add post on dependency variables

### PR DESCRIPTION
**Pull request by AltGr**

### CODE REVIEW

File: opam-extended-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+#### Dependency flags
+
+Dependency flags, as used in 1.2, are no longer needed, and are replaced by
+variables that can appear anywhere in the version specification. The following
+variables are typically useful there:
+
+- `with-test`, `with-doc`: replace the `test` and `doc` dependency flags, and
+  `true` when the package's tests or documentation have been requested
```
Comment (dbuenzli) [inline opam-extended-dependencies.md]: and *are* `true`

File: opam-extended-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- likewise, `build` behaves similarly as before, limiting the dependency to a
+  "build-dependency", implying that the package won't need to be rebuilt if it
```
Comment (dbuenzli) [inline opam-extended-dependencies.md]: if *the dependency* changes

File: opam-extended-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  changes
+- `dev`: this boolean variable holds `true` on "development" packages, that is,
+  packages that are bound to a non-stable source (a version control system, or
+  if the package is pinned without a source checksum specified). `dev` sources
```
Comment (dbuenzli) [inline opam-extended-dependencies.md]: I'm not sure I understand the "if the package is pinned without a source checksum specified" condition, what is it concretly ?

**Comment by dbuenzli** [comment]
Since this is about dependencies, one thing you could add as a tidbit is that `available:` is gone (if I'm not mistaken) and that ocaml versions are not expressed as a package constraint.

**Comment by Drup** [comment]
This is very nice. Maybe you could take the example of a package with a test dependencies and see how it is translated concretely (since it's a very common use case).

**Pull request merged**